### ISSUE TITLE
GeneralLinearModelAlgorithm mishandles user-specified scale parameter when normalize is True

### ISSUE THREAD

**Issue by josephmure** [openturns/openturns#1407]
When the scale parameter is inactive, it means that the user already knows its value.
This value applies to the user-specified `inputSample`, but the `GeneralLinearModelAlgorithm` assumes that it applies to the normalized `inputSample` used during parameter optimization.
The following script illustrates the problem:

```
import numpy as np
import openturns as ot

# Using the example from the KrigingAlgorithm doc.
f = ot.SymbolicFunction(['x'],  ['x * sin(x)'])
sampleX = [[1.0], [2.0], [3.0], [4.0], [5.0], [6.0], [7.0], [8.0]]
sampleY = f(sampleX)
basis = ot.Basis([ot.SymbolicFunction(['x'], ['x']), ot.SymbolicFunction(['x'], ['x^2'])])

# Specify the covariance model. We only want to estimate the scale parameter.
covarianceModel = ot.SquaredExponential([1.0], [16.0])
covarianceModel.setActiveParameter([0])

# Run the Kriging algorithm. According to the doc, the GeneralLinearModelAlgorithm normalizes the input sampleX.
algo = ot.KrigingAlgorithm(sampleX, sampleY, covarianceModel, basis)
algo.run()

# Get the KrigingResult.
result = algo.getResult()

# Get the normalzed input sample which was used by the GeneralLinearModelAlgorithm optimization process.
normalized_sampleX = result.getTransformation()(sampleX)
print("normalized_sampleX = ")
print(normalized_sampleX)
print()

# Get the optimized scale. It fits normalized_sampleX, not the original sampleX.
scale = result.getCovarianceModel().getScale()
print("Optimized scale for normalized_sampleX = ", scale)
print()

# Apply the optimized covariance model to normalized_sampleX.
cov_matrix = result.getCovarianceModel().discretize(normalized_sampleX)
print("The covariance matrix of the Gaussian vector sampleY is assumed to be a realization of is: ")
print(cov_matrix)
print()

# Take normalized_sampleX and stretch it out by the factor 50.
sampleX_times50 = ot.Sample(np.array(normalized_sampleX) * 50)
print("sampleX_times50 is 50 times the normalized_sampleX : ")
print(sampleX_times50)
print()

# Logically, the optimal scale parameter for sampleX_times50 should be the optimized scale multiplied by 50.
covarianceModel.setScale(scale * 50)

# I know everything about the optimal CovarianceModel. No need to optimize anything.
covarianceModel.setActiveParameter([])

# Let us run the KrigingAlgorithm on sampleX_times50 and get the KrigingResult.
algo_times50 = ot.KrigingAlgorithm(sampleX_times50, sampleY, covarianceModel, basis)
algo_times50.run()
result_times50 = algo_times50.getResult()
normalized_sampleX_from_sampleX_times50 = result_times50.getTransformation()(sampleX_times50)
print("The normalized version of sampleX_times50 is equal to the previously computed normalized_sampleX: ")
print(normalized_sampleX_from_sampleX_times50)
print()

# You would assume that the KrigingAlgorithm knows that the user-supplied scale (scale*50) 
# applies to the user-supplied input sample (sampleX_times50) and not to the normalized sample. 
# You would be wrong.
scale_times50 = result_times50.getCovarianceModel().getScale()
print("When the scale parameter is inactive, it is not adjusted to normalized_sampleX: ", scale_times50)
print()

# Any user trying to extract the covariance matrix is in for a shock.
cov_matrix_times50 = result_times50.getCovarianceModel().discretize(normalized_sampleX_from_sampleX_times50)
print("The covariance matrix is therefore false: ")
print(cov_matrix_times50)
```
Running this script produces:
```
normalized_sampleX = 
0 : [ -1.42887  ]
1 : [ -1.02062  ]
2 : [ -0.612372 ]
3 : [ -0.204124 ]
4 : [  0.204124 ]
5 : [  0.612372 ]
6 : [  1.02062  ]
7 : [  1.42887  ]

Optimized scale for normalized_sampleX =  [1.10326]

The covariance matrix of the Gaussian vector sampleY is assumed to be a realization of is: 
8x8
[[ 256      239.06   194.673  138.242   85.6063  46.228   21.769    8.9393 ]
 [ 239.06   256      239.06   194.673  138.242   85.6063  46.228   21.769  ]
 [ 194.673  239.06   256      239.06   194.673  138.242   85.6063  46.228  ]
 [ 138.242  194.673  239.06   256      239.06   194.673  138.242   85.6063 ]
 [  85.6063 138.242  194.673  239.06   256      239.06   194.673  138.242  ]
 [  46.228   85.6063 138.242  194.673  239.06   256      239.06   194.673  ]
 [  21.769   46.228   85.6063 138.242  194.673  239.06   256      239.06   ]
 [   8.9393  21.769   46.228   85.6063 138.242  194.673  239.06   256      ]]

sampleX_times50 is 50 times the normalized_sampleX : 
0 : [ -71.4435 ]
1 : [ -51.031  ]
2 : [ -30.6186 ]
3 : [ -10.2062 ]
4 : [  10.2062 ]
5 : [  30.6186 ]
6 : [  51.031  ]
7 : [  71.4435 ]

The normalized version of sampleX_times50 is equal to the previously computed normalized_sampleX: 
0 : [ -1.42887  ]
1 : [ -1.02062  ]
2 : [ -0.612372 ]
3 : [ -0.204124 ]
4 : [  0.204124 ]
5 : [  0.612372 ]
6 : [  1.02062  ]
7 : [  1.42887  ]

When the scale parameter is inactive, it is not adjusted to normalized_sampleX:  [55.1632]

The covariance matrix is therefore false: 
8x8
[[ 256     255.993 255.972 255.937 255.888 255.825 255.748 255.657 ]
 [ 255.993 256     255.993 255.972 255.937 255.888 255.825 255.748 ]
 [ 255.972 255.993 256     255.993 255.972 255.937 255.888 255.825 ]
 [ 255.937 255.972 255.993 256     255.993 255.972 255.937 255.888 ]
 [ 255.888 255.937 255.972 255.993 256     255.993 255.972 255.937 ]
 [ 255.825 255.888 255.937 255.972 255.993 256     255.993 255.972 ]
 [ 255.748 255.825 255.888 255.937 255.972 255.993 256     255.993 ]
 [ 255.657 255.748 255.825 255.888 255.937 255.972 255.993 256     ]]
```
Obviously, the user-specified scale parameter should be normalized like the `inputSample` even if it is inactive. I wonder if perhaps the method `getInputSample()` from the `KrigingResult` should yield the normalized `inputSample` in order to fit the `CovarianceModel` that can be obtained from the `getCovarianceModel()` method. Currently, `getInputSample()` yields the original inputSample.

**Comment by jschueller** [comment]
TODO: get rid of normalization in kriging

**Comment by mbaudin47** [comment]
For reference here is how this is handled in Scikit:

https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.normalize.html

In Matlab:

https://www.mathworks.com/help/matlab/ref/double.normalize.html

In otsvm:

https://github.com/openturns/otsvm/blob/7932689cebd3d091a84b9832bc80d739631e518d/lib/src/otsvm/LibSVM.hxx#L114

**Comment by sofianehaddad** [comment]
Let us come back to solve this issue.
@regislebrun @mbaudin47 @josephmure @jschueller what do you have in mind :

 - Drop the `normalize` input parameter ==> no more transformation handling
or
 - Replace this with new ctors involving a `transformation`. 
In that case, `KrigingAlgorithm(inputSample, outputSample, covarianceModel, basis, normalize)`
becomes `KrigingAlgorithm(inputSample, inputTransformation, outputSample, covarianceModel, basis)` ==> we might have a transformation, which is propagated to `KrigingResult` and so but it becomes the responsibility of user. 

Notice also that we should apply same changes to `GeneralLinearModelAlgorithm`
